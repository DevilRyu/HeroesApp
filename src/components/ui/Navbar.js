import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

    const {user:{name}, dispatch} = useContext(AuthContext);

    const history= useHistory();

    const handleLogout = () => {

        history.replace('/HeroesApp/login');
        
        dispatch({
            type: types.logout
        });
       
    };

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark ">
            <div className="container-fluid">
                <Link
                    className="navbar-brand"
                    to="/HeroesApp"
                >
                    Heroes
                </Link>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse" id="navbarSupportedContent" >
                    <ul className="navbar-nav me-auto mb-2 mb-lg-0">

                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/HeroesApp/marvel"
                        >
                            Marvel
                        </NavLink>

                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/HeroesApp/dc"
                        >
                            DC
                        </NavLink>
                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/HeroesApp/search"
                        >
                            Search
                        </NavLink>
                    </ul>
                    <div className="d-flex">
                        <ul className="navbar-nav ml-auto">
                            <span className="nav-item nav-link text-info">
                                {name}
                            </span>
                            <button
                                className="btn nav-item nav-link"
                                onClick={handleLogout}
                            >
                                Logout
                            </button>
                        </ul>
                    </div>
                </div>


            </div>

        </nav>
    )
}