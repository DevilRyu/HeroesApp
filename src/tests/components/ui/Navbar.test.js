import React from 'react';
import { mount } from 'enzyme';
import { Navbar } from '../../../components/ui/Navbar';
import { MemoryRouter, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';
import '@testing-library/jest-dom';

describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Diego'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach( () => {
        jest.clearAllMocks();
    });

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Diego');

    });

    test('debe de llamar logout y el usar history', () => {

        wrapper.find('button').find('.btn').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });
        
        expect(historyMock.replace).toHaveBeenCalledWith('/HeroesApp/login')

    });

});