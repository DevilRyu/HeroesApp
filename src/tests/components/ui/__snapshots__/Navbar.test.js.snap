// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark "
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Heroes
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Search
        </a>
      </ul>
      <div
        className="d-flex"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <span
            className="nav-item nav-link text-info"
          >
            Diego
          </span>
          <button
            className="btn nav-item nav-link"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
